homepage/index.html

```diff
@@ -6,825 +6,741 @@
-                    .then((response) => response.json())
-                    .then((response) => {
-                      if (!response.eligible) {
-                        hideElements('eligible')
-                        revealElements('ineligible')
-                      }
-                      return response.eligible
-                    }
-            );
-          }
-
-          function setUserState() {
-              if (!window.fetch) {
-                  revealElements('login-signup');
-                  return;
-              }
-              return window.fetch(URLS.WHOAMI, {
-                  method: 'GET',
-                  credentials: 'include'
-              }).then(function(resp) {
-                  resp.json().then(function(user) {
-                      if (user.id) {
-                          revealElements('start-using');
-                      } else {
-                          revealElements('login-signup');
-                      }
-                  });
-              }).catch(function() {
-                  revealElements('login-signup');
-              });
-          }
-
-          function rewriteLinks(klass, href) {
-              var links = document.querySelectorAll('.link.' + klass);
-              for (var i = 0; i < links.length; i++) {
-                  links[i].href = href;
-              }
-          }
-          setupStepActions(steps);
-          rewriteLinks('sample-study', URLS.SAMPLE_STUDY)
-          rewriteLinks('login', URLS.ACCOUNTS + '/i/login?r=' + URLS.KINETIC);
-          rewriteLinks('signup', URLS.ACCOUNTS + '/i/signup?r=' + URLS.KINETIC);
-          rewriteLinks('start', URLS.KINETIC);
-          checkEligibility().then(isEligible => isEligible && setUserState())
-
-          function onPageUnload() {
-              if (previousFavIcon) {
-                  previousFavIcon.el.href = previousFavIcon.href;
-              }
-              delete window.KINETIC;
-          }
-
-          var observer = new MutationObserver(function(mutations_list) {
-              mutations_list.forEach(function(mutation) {
-                  mutation.removedNodes.forEach(function(removed_node) {
-                      if(removed_node.className === 'general page') {
-                          onPageUnload();
-                          observer.disconnect();
-                      }
-                  });
-              });
+        const assignLink = (className, link) => {
+          const elements = document.querySelectorAll("." + className);
+
+          elements.forEach((element) => {
+            element.onclick = () => {
+              console.log("Clicked");
+              window.location.href = link;
+            };
           });
-          observer.observe(document, { subtree: true, childList: true });
-          window.KINETIC.postScriptRan = true;
-      }
+        };
+      document.addEventListener('DOMContentLoaded', () => {
```

Comment:
is there a reason we can't just use the `href` attribute on these links wherever they're defined in html?

Reply:
I did it in javascript to make it uniform. I'll add the href tags